rustup-init.ps1

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env powershell
+
+# This is just a little script that can be downloaded from the internet to
+# install rustup. It just does platform detection, downloads the installer and
+# runs it.
+
+#XXX: If you change anything here, please make the same changes in setup_mode.rs
+function Write-Usage {
+    Write-Host @'
+rustup-init 1.18.3 (435397f48 2019-05-22)
+The installer for rustup
+
+USAGE:
+    rustup-init.exe [FLAGS] [OPTIONS]
+
+FLAGS:
+    -v, --verbose           Enable verbose output
+    -y                      Disable confirmation prompt.
```

Comment:
Confirmation is already implemented in the default cmdlet binding. Do not re-implement.
https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_commonparameters?view=powershell-6

https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_Preference_Variables?view=powershell-6#confirmpreference

Use `$ConfirmPreference` instead

Reply:
I think you've partly misunderstood - these arguments are passed to the downloaded `rustup-init.exe`

Reply:
True, but these are also the commands that the PowerShell script is expecting. The PowerShell script should be taking the PowerShell style parameters and crafting the call to `rustup-init.exe`.  Currently it's passing verbatim (linux style parameters) with a short circuit for help.

Reply:
Like @glennsarti said, Verbose and Confirm parameters are provided by the cmdlet binding, but you still have to implement the logic. In this case, reading the PowerShell `$ConfirmPreference` and translating that to the appropriate flag for `rustup-init.exe`. This allows a PowerShell users to set `$VerbosePreference` and `$ConfirmPreference` higher in the call stack or globally in their session without having to know the internals of `rustup-init`.

Reply:
While I have a weak preference for ensuring that the command line syntax is the same for `rustup-init.ps1` as it will be for `rustup-init.exe` I am not motivated to block this rework if it's over-all desirability to Windows developers is high.

It's worth noting though that this is, in general, only going to be used in CI situations since most developers will simply download `rustup-init.exe` and double-click on it.

Reply:
The `-y` looks to be the only argument that would be functionally different. Granted, the PowerShell `Confirm` parameter usage is a bit counter-intuitive.

Reply:
> It's worth noting though that this is, in general, only going to be used in CI situations since most developers will simply download rustup-init.exe and double-click on it.

Maybe initially, but I strongly disagree moving forward. People *will* find it's much easier to use a PowerShell script than go-to-website-download-exe-double-click-next-next-finish. 

Applying the same argument to the `https://sh.rustup.rs` script.  Why do you have it? Is that just for CI? nope.  Users found that useful instead of go-to-website-download-rpm/deb/dmg-run-that

Reply:
I think the important thing here is that:
 - the documentation on how to use this script is clear
 - it interacts with the chained rustup-init appropriately: that is, -y is passed to rustup-init if the user selected that

I don't think it matters whether users will find the ps1 script better or worse, or more or less appealing than the website: we'll only find that out by measuring what happens after this is merged, so lets focus on making that happen in a way that is sustainable and works well in the Powershell ecosystem (which I know nothing about, and shall be refraining from adding chaos to the review because of that).